src/main/java/com/scottlogic/librarygradproject/Book.java

```diff
@@ -86,7 +87,7 @@ public boolean equals(Object o) {
 
     @Override
     public int hashCode() {
-        int result = id;
+        int result = (int) id;
```

Comment:
If we ever have to use the hashCode function, could we get overflow when casting?

Reply:
Yeah potentially but i dont think its a problem. If you look at what the rest of the code does it multiplies the result a number of times so I think overflow is expected. Ultimately we need to return an int value which will always result in an overflow if id is high.